creusot-contracts/src/std/ord.rs

```diff
@@ -0,0 +1,101 @@
+use crate as creusot_contracts;
+use creusot_contracts_proc::*;
+
+use std::cmp::Ordering;
+
+pub trait Ord {
+    #[logic]
+    fn cmp_log(self, _: Self) -> Ordering;
+
+    #[ensures(result === self.cmp_log(*o))]
+    fn cmp(&self, o: &Self) -> Ordering;
+
+    #[logic]
+    fn le_log(self, o: Self) -> bool {
+        match self.cmp_log(o) {
+            Ordering::Greater => false,
+            _ => true,
+        }
+    }
+
+    #[ensures(result === self.le_log(*o))]
+    fn le(&self, o: &Self) -> bool {
+        match self.cmp(o) {
+            Ordering::Greater => false,
+            _ => true,
+        }
+    }
+
+    #[logic]
+    fn ge_log(self, o: Self) -> bool {
+        match self.cmp_log(o) {
+            Ordering::Less => false,
+            _ => true,
+        }
+    }
+
+    #[ensures(result === self.ge_log(*o))]
+    fn ge(&self, o: &Self) -> bool {
+        match self.cmp(o) {
+            Ordering::Less => false,
+            _ => true,
+        }
+    }
+
+    #[logic]
+    fn gt_log(self, o: Self) -> bool {
+        match self.cmp_log(o) {
+            Ordering::Greater => true,
+            _ => false,
+        }
+    }
+
+    #[ensures(result === self.gt_log(*o))]
+    fn gt(&self, o: &Self) -> bool {
+        match self.cmp(o) {
+            Ordering::Greater => true,
+            _ => false,
+        }
+    }
+
+    #[logic]
+    fn lt_log(self, o: Self) -> bool {
+        match self.cmp_log(o) {
+            Ordering::Less => true,
+            _ => false,
+        }
+    }
+
+    #[ensures(result === self.lt_log(*o))]
+    fn lt(&self, o: &Self) -> bool {
+        match self.cmp(o) {
+            Ordering::Less => true,
+            _ => false,
+        }
+    }
+
+    #[logic]
+    #[ensures(x.cmp_log(x) === Ordering::Equal)]
+    fn refl(x: Self);
+
+    #[logic]
+    #[requires(x.cmp_log(y) === o)]
+    #[requires(y.cmp_log(x) === o)]
+    #[ensures(x.cmp_log(z) === o)]
+    fn trans(x: Self, y: Self, z: Self, o: Ordering);
+
+    #[logic]
+    #[requires(x.cmp_log(y) === Ordering::Less)]
+    #[ensures(y.cmp_log(x) === Ordering::Greater)]
+    fn antisym1(x: Self, y: Self);
+
+    #[logic]
+    #[requires(x.cmp_log(y) === Ordering::Greater)]
+    #[ensures(y.cmp_log(x) === Ordering::Less)]
+    fn antisym2(x: Self, y: Self);
+
+    #[logic]
+    #[requires(x.cmp_log(y) === Ordering::Equal)]
+    #[ensures(y.cmp_log(x) === Ordering::Equal)]
+    fn symmetry(x: Self, y: Self);
```

Comment:
Same thing here: this should not be called `symmetry`. The whole point of order relations is that they are *not* symetric.